src/codec.ts

```diff
@@ -27,47 +25,29 @@ export class NoiseHandshakeMessage extends MessageV0 implements Message {
  * Used in the pairing object for encoding the messages exchanged
  * during the handshake process
  */
-export class NoiseHandshakeEncoder implements Encoder {
+export class NoiseHandshakeEncoder extends Encoder {
```

Comment:
The aim is for this class to be usable as an `Encoder`, right? hence, I would expect it to `implements IEncoder`.
I assume this is not inherited from `Encoder` because the present implementation could make it differ from the `IEncoder` interface.

note: the `extends Encoder` can remain, i am only talking about **adding** the `implements` statement.

Reply:
Added usage of `implements` and left prev implementation of the `NoiseHandshakeEncoder` for now.